### PR TITLE
[jsk_aibo_robot][aibo_selenium_ros] add package

### PR DESCRIPTION
**Pull request by sktometometo**
Add aibo_selenium_ros

### CODE REVIEW

File: jsk_aibo_robot/aibo_selenium_ros/node_scripts/main.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    input('Press Enter when logging if completed.')
+    interface.initialized = True
+
+  if not interface.initialized:
+    rospy.logerr('Initialization failed.')
+    input('Press Enter when logging if completed.')
+
+  interface.start_watching()
+  while True:
+    image_rgb = interface.get_current_image()
```
Comment (k-okada) [inline jsk_aibo_robot/aibo_selenium_ros/node_scripts/main.py:46]: nice !  can you also publish compressed image? or create launch file that also start compressed_image_transport.

Comment (sktometometo) [reply jsk_aibo_robot/aibo_selenium_ros/node_scripts/main.py:47]: Added.

File: jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
+import base64
+import io
+import logging
+import random
+import time
+
+import selenium
+from imageio import imread
```
Comment (k-okada) [inline jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py:8]: add iamgeio to pcakge.xml or requirements.txt

File: jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  def start_watching(self, timeout=5.):
+
+    try:
+      for b in self.driver.find_elements(By.TAG_NAME, value='button'):
+        if b.accessible_name == '今なに見てる？' and b.is_displayed():
+          b.click()
+          break
+      time.sleep(10.)
+    except Exception as e:
+      logger.logwarn('Got an error {}'.format(e))
```
Comment (k-okada) [inline jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py:126]: warn? warning?

File: jsk_aibo_robot/aibo_selenium_ros/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+This package provides a ROS Node which retieve head camera images from web browser and publish it as ROS message.
+
+## How to setup
+
+**Currently this node is only tested with ROS noetic.**
+
+First, Install [Google chrome](https://www.google.com/intl/ja_jp/chrome/) to your PC.
+
+Second, Download [suitable version of chrome webdriver to Google chrome installed](https://chromedriver.chromium.org/downloads) and place it somewhere.
```
Comment (k-okada) [inline jsk_aibo_robot/aibo_selenium_ros/README.md:11]: is there any reason you do not use system `chromium-browser` and`chromium-chromedriver`

File: jsk_aibo_robot/aibo_selenium_ros/node_scripts/main.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  rospy.init_node('aibo_selenium_ros')
+
+  pub_raw = rospy.Publisher('~image', Image, queue_size=1)
+  pub_compressed = rospy.Publisher('~image/compressed', CompressedImage, queue_size=1)
+  cv_bridge = CvBridge()
+
+  logging.basicConfig(level=logging.INFO)
+  logger = logging.getLogger(__name__)
+
+  webdriver = rospy.get_param('~webdriver')
```
Comment (k-okada) [inline jsk_aibo_robot/aibo_selenium_ros/node_scripts/main.py:25]: you can use `/usr/bin/chromedriver` as a default value.

File: jsk_aibo_robot/aibo_selenium_ros/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(aibo_selenium_ros)
+
+find_package(catkin REQUIRED)
+
+catkin_python_setup()
+
+catkin_package()
```
Comment (k-okada) [inline jsk_aibo_robot/aibo_selenium_ros/CMakeLists.txt:8]: add somethign like

```
file(GLOB NODE_SCRIPTS_FILES node_scripts/*.py)

catkin_install_python(
  PROGRAMS ${NODE_SCRIPTS_FILES}
  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
)
```

File: jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py (hunk 2 above)
Comment (sktometometo) [reply jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py:8]: Added

File: jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py (hunk 3 above)
Comment (sktometometo) [reply jsk_aibo_robot/aibo_selenium_ros/python/aibo_selenium_ros/__init__.py]: Fixed.

File: jsk_aibo_robot/aibo_selenium_ros/README.md (hunk 4 above)
Comment (sktometometo) [reply jsk_aibo_robot/aibo_selenium_ros/README.md]: Changed to chromium and use it.

File: jsk_aibo_robot/aibo_selenium_ros/node_scripts/main.py (hunk 5 above)
Comment (sktometometo) [reply jsk_aibo_robot/aibo_selenium_ros/node_scripts/main.py]: Set.

File: jsk_aibo_robot/aibo_selenium_ros/CMakeLists.txt (hunk 6 above)
Comment (sktometometo) [reply jsk_aibo_robot/aibo_selenium_ros/CMakeLists.txt:8]: Added.

File: jsk_aibo_robot/aibo_selenium_ros/package.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>aibo_selenium_ros</name>
+  <version>0.0.0</version>
```
[the remark this answers is not in the document]
Comment (sktometometo) [reply jsk_aibo_robot/aibo_selenium_ros/package.xml]: Set

**Comment by sktometometo** [comment]
`chromedriver` rosdep entry can be resolved after https://github.com/ros/rosdistro/pull/37167 is merged.

**Comment by sktometometo** [comment]
Moved to https://github.com/jsk-ros-pkg/jsk_robot/pull/1852/files.

**Pull request closed without merging**